### ISSUE TITLE
Need a Linux version

### ISSUE THREAD

**Issue by coreybruce** [7D2DModLauncher/ThickInstaller#2]
Hey just wanted to ask if you could please make a Linux version as 7D2D is also on Linux and cannot use this launcher for the game to install/manage mods and can't use the installer to install it.

We can run .net applications in Linux with mono which is .net for Linux but I get this error running the launcher. 

http://i.imgur.com/1XBRED9.png

**Comment by synsa** [comment]
Considering that 7D2D works on Win/OSX/Linux, a version for linux would be nice to have.

**Comment by laegnur** [comment]
If the game is native on different platforms. The mods must be multi platform also.

**Comment by coreybruce** [comment]
Yes but the mod launcher isn't, it's only for Windows..

**Comment by SphereII** [comment]
New beta version of the mod launcher is available here for Linux platform: https://github.com/SphereII/7D2DModLauncher/releases/tag/Latest

While the mod launcher is multi-platform, some mods may or may not work 100% properly under Linux due to texture anomalies.

**Comment by coreybruce** [comment]
> New beta version of the mod launcher is available here for Linux platform: https://github.com/SphereII/7D2DModLauncher/releases/tag/Latest
> 
> While the mod launcher is multi-platform, some mods may or may not work 100% properly under Linux due to texture anomalies.

Cool, when will this be out of beta? going to be trying it now because I wanna play the DarknessFalls mod with a friend.

Also you should add this to the AUR and flatpak or even a deb file for people on Linux.

**Comment by SphereII** [comment]
Its fully functional, but I'm still improving the UI and error handling. I'll probably release it for A20.

There are so many Linux distributions, I'm not sure how practical it would be for me to maintain packages. 

The mod launcher is auto-updated however on start up.

**Comment by coreybruce** [comment]
on the launcher I am unable to launch the game with the darkness falls mod.

I had to manually launch ./7DaysToDie.sh from inside the mod folder to get the game working.

As for maintaining it shouldn't be difficult since all Linux distros all have the same dependencies even if some are under a slight different name it's if not easier because every package manager can be told to install what ever dependencies are needed when you install them unlike on Windows where you need to manually download and install anything extra yourself manually. If you also provide a flatpak option than maintaining for different distro's are as simple as supplying one OS but a good place to start for the easiest distribution is the AUR.  

I could even easily make a AUR package for this as I do software development myself but it would be better if you did it yourself instead.

**Comment by SphereII** [comment]
Can you attach a log file of the mod launcher?

**Comment by coreybruce** [comment]
Sure, where does it usually store the log for it?

**Comment by SphereII** [comment]
Log Files: 

    ~/.config/unity3d/SphereII Software/ModLauncher/Player.log

**Comment by coreybruce** [comment]
I see the issue, this might need to be added to the code to avoid this issue but because my drive name has a space it thinks it's 2 separate locations

```
Launching Game from /mnt/Game Drive/7d2d-mods/Alpha19/Darkness_Falls/Darkness_Falls/

No Log File exists, backup not possible: /mnt/Game Drive/7d2d-mods/Alpha19/Darkness_Falls/Darkness_Falls/7DaysToDie_Data/output_log.txt

/usr/bin/chmod: cannot access '/mnt/Game': No such file or directory
/usr/bin/chmod: cannot access 'Drive/7d2d-mods/Alpha19/Darkness_Falls/Darkness_Falls/7DaysToDie.x86_64': No such file or directory
Starting /mnt/Game Drive/7d2d-mods/Alpha19/Darkness_Falls/Darkness_Falls/7DaysToDie.x86_64  -logfile /mnt/Game Drive/7d2d-mods/Alpha19/Darkness_Falls/Darkness_Falls/7DaysToDie_Data/output_log.txt "-savegamefolder=/mnt/Game Drive/7d2d-mods/Saves/Darkness Falls/Darkness Falls" -nogs -noeac

Found path: /mnt/Game Drive/7d2d-mods/Alpha19/Darkness_Falls/Darkness_Falls/7DaysToDie.x86_64
Unable to open log file, exiting.
Setting up 3 worker threads for Enlighten.
  Thread -> id: 7f3387e8d640 -> priority: 1 
  Thread -> id: 7f338768c640 -> priority: 1 
  Thread -> id: 7f3386e8b640 -> priority: 1 
Waiting on threads to park on joinable thread list timed out.
```

Also every time I open the launcher it want's to do a 300mb update, the current version is 3.0.3.14. 

Here is a full log

https://gist.github.com/coreybruce/6d67529b9201c266edf259b0397ba953


Looks like path reading needs to be fixed even if a path has a space in it which on Linux you can, I did do a workaround by simlinking the mod launcher to .config than making a mod folder inside that and simlinked my darkness falls modded game inside that folder and told it to read that path which is 

 ```/home/corey/.config/7d2dmodlauncher/```

and 

```/home/corey/.config/7d2dmodlauncher/mods/```

But obviously this isn't a solution and more of work around to something that needs to be fixed. :)

**Comment by SphereII** [comment]
Thank you, I'll take a look and see what I can do. Seems like it's just a small part that is struggling with the spaces, since the rest of the install worked well.

For what its worth, under the old Windows Mod Launcher client, spaces weren't supported; they were switched to replace spaces with underscores since some commands worked or not.

**Comment by coreybruce** [comment]
Will the fix for it re-downloading the update every time I open the launcher be fixed?

**Comment by SphereII** [comment]
I will have to investigate that more. It does not do that for me under Ubuntu or SuSe. I have heard it dong that on Windows as well, but seems inconsistent. It's a third-party library that handles all that logic.

If you look for a patcher.config under 7D2DModLauncher_Data, you could try changing the <Check>true</Check> to <Check>false</Check> and see if it still does it.

**Comment by coreybruce** [comment]
"I will have to investigate that more. It does not do that for me under Ubuntu or SuSe. I have heard it dong that on Windows as well, but seems inconsistent. It's a third-party library that handles all that logic."

This is what it looks like everytime I open it: 
![Screenshot_2021-09-16_17-30-10](https://user-images.githubusercontent.com/8393780/133570067-61f6c64d-ce1e-43a8-95dc-a98dc1151e53.png)

"If you look for a patcher.config under 7D2DModLauncher_Data, you could try changing the true to false and see if it still does it."

Yeah that stops it from re-downloading the update every time.
